sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BeamRowToStorageApiProto.java

```diff
@@ -291,9 +307,17 @@ private static Object toProtoValue(
         if (arrayElementType == null) {
           throw new RuntimeException("Unexpected null element type!");
         }
-        return list.stream()
-            .map(v -> toProtoValue(fieldDescriptor, arrayElementType, v))
-            .collect(Collectors.toList());
+        boolean shouldFlatMap =
```

Comment:
This looks like it supports one level of nested collection. Is that an intended limit?

Reply:
Also, can you add a comment to that effect?

Reply:
Yes I will add the comment. 

Treating the recursive collection flattening should help to cover the other collection types. This PR focus on the special case of having a MAP, or ARRAY<MAP> as the type of a field in a Row.

Reply:
Removed the flattening of nested container types.